### DISCUSSION TITLE
Bug relationships: Have one bug depend upon another bug.

### DISCUSSION THREAD

**Discussion by dotancohen** [git-bug/git-bug#598] [Q&A]
I can find no documentation regarding bug relationships. How does one set a bug to be dependent upon another bug in git-bug? Thank you.

**Comment by claudioantonio** [comment]
Hi @dotancohen , interesting question. Do You have a Jira project with issues linked to others (caused by, relates to, is dependent of, etc.) and wants to know if those links will be imported to git-bug?

The GraphQL API does not provide a mutator for this feature today.

**Reply by dotancohen** [reply]
When I was using Jira at a former company we used these relationships extensively.

Right now I'm starting a personal project and considering using git-bug. I do not have a Jira project to import, but I would require these relationships from any issue tracker that I use.

**Reply by claudioantonio** [reply]
Hi @dotancohen 
1.  I found a 2019 comment related to this topic: https://github.com/MichaelMure/git-bug/pull/91#issuecomment-452438615
2. Do You think a single relationship like Issue X "relates to" Issue Y (so one issue would have a link to each other) would be sufficient?

@MichaelMure in the comment I mentioned before You said:
> It's a tricky subject, both technically and for the UX. I'll have to make sure that it fits nicely with other features and workflows.

Could You please talk more about this subject?

**Reply by GlancingMind** [reply]
Maybe links between Issues could be introduce via [Webmentions](https://www.w3.org/TR/webmention).
This would also allow to mention Issues across multiple hosted Git-Bug instances.
But I'm not sure about the compatibility with the CLI and TerminalUI.

**Reply by dotancohen** [reply]
In my opinion, a single metadata field "Depends on" could be attached to an issue. This would be displayed when viewing the issue.

Also, the bugs that depend upon a bug could be shown when viewing a bug either by default or with a `--show-dependencies` flag, as it would require parsing all other bugs' metadata for this field.

I would also suggest reserving a `dependencyType` field, required if the `dependsOn` field is in use. Though I personally do not have a use for it, I do understand that Jira and other platforms allow for different types of dependencies and people use them.
